fast-list.js

```diff
@@ -389,10 +390,21 @@ FastList.prototype = {
    */
 
   handleScroll: function(evt) {
+    window.clearTimeout(this._scrollSentinel);
+
     this.updateRenderingWindow();
 
     if (this.geometry.busy) debug('[x] ---------- faaaaassssstttt');
     else this.render();
+
+    if (this.geometry.idle) return;
+
+    // We just did a partial rendering and need to make sure it won't
+    // get stuck this way if the scrolling comes to a hard stop.
+    this._scrollSentinel = setTimeout(() => {
+      this.updateRenderingWindow(true);
+      this.render();
+    }, 200);
```

Comment:
Well, the `200` duration was pretty carefully chosen yes.
Not sure if the constant clearTimeout/seTimeout would show up in profiles, but I'm not sure how we can do without it.

Good news is bug 1172171 has an assignee :)